### ISSUE TITLE
Unable to enumerate due to an error: vim.CustomFieldsManager.StringValue is not JSON serializable

### ISSUE THREAD

**Issue by mrthomaskim** [vmware/pyvmomi#768]
I am getting an error message when trying to enumerate the list of all VMs using ansible-inventory:
ERROR! Unexpected Exception, this is probably a bug: Object of type vim.CustomFieldsManager.StringValue is not JSON serializable

We do utilize custom attributes sections for tagging purpose

Below is the full debug:
```
$ ansible-inventory  --list -i dynamic.vmware.yml -vvvv
ansible-inventory 2.7.8
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/Users/tkim/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/tkim/.local/share/virtualenvs/ansible-sskOqjbK/lib/python3.7/site-packages/ansible
  executable location = /Users/tkim/.local/share/virtualenvs/ansible-sskOqjbK/bin/ansible-inventory
  python version = 3.7.2 (default, Feb 12 2019, 08:15:36) [Clang 10.0.0 (clang-1000.11.45.5)]
Using /etc/ansible/ansible.cfg as config file
setting up inventory plugins
Parsed /Users/tkim/git/ccc/sre/ansible/dynamic.vmware.yml inventory source with vmware_vm_inventory plugin
ERROR! Unexpected Exception, this is probably a bug: Object of type vim.CustomFieldsManager.StringValue is not JSON serializable
the full traceback was:

Traceback (most recent call last):
  File "/Users/tkim/.local/share/virtualenvs/ansible-sskOqjbK/bin/ansible-inventory", line 118, in <module>
    exit_code = cli.run()
  File "/Users/tkim/.local/share/virtualenvs/ansible-sskOqjbK/lib/python3.7/site-packages/ansible/cli/inventory.py", line 181, in run
    results = self.dump(results)
  File "/Users/tkim/.local/share/virtualenvs/ansible-sskOqjbK/lib/python3.7/site-packages/ansible/cli/inventory.py", line 199, in dump
    results = json.dumps(stuff, cls=AnsibleJSONEncoder, sort_keys=True, indent=4)
  File "/usr/local/Cellar/python/3.7.2_2/Frameworks/Python.framework/Versions/3.7/lib/python3.7/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
  File "/usr/local/Cellar/python/3.7.2_2/Frameworks/Python.framework/Versions/3.7/lib/python3.7/json/encoder.py", line 201, in encode
    chunks = list(chunks)
  File "/usr/local/Cellar/python/3.7.2_2/Frameworks/Python.framework/Versions/3.7/lib/python3.7/json/encoder.py", line 431, in _iterencode
    yield from _iterencode_dict(o, _current_indent_level)
  File "/usr/local/Cellar/python/3.7.2_2/Frameworks/Python.framework/Versions/3.7/lib/python3.7/json/encoder.py", line 405, in _iterencode_dict
    yield from chunks
  File "/usr/local/Cellar/python/3.7.2_2/Frameworks/Python.framework/Versions/3.7/lib/python3.7/json/encoder.py", line 405, in _iterencode_dict
    yield from chunks
  File "/usr/local/Cellar/python/3.7.2_2/Frameworks/Python.framework/Versions/3.7/lib/python3.7/json/encoder.py", line 405, in _iterencode_dict
    yield from chunks
  [Previous line repeated 1 more time]
  File "/usr/local/Cellar/python/3.7.2_2/Frameworks/Python.framework/Versions/3.7/lib/python3.7/json/encoder.py", line 325, in _iterencode_list
    yield from chunks
  File "/usr/local/Cellar/python/3.7.2_2/Frameworks/Python.framework/Versions/3.7/lib/python3.7/json/encoder.py", line 438, in _iterencode
    o = _default(o)
  File "/Users/tkim/.local/share/virtualenvs/ansible-sskOqjbK/lib/python3.7/site-packages/ansible/parsing/ajson.py", line 66, in default
    value = super(AnsibleJSONEncoder, self).default(o)
  File "/usr/local/Cellar/python/3.7.2_2/Frameworks/Python.framework/Versions/3.7/lib/python3.7/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type vim.CustomFieldsManager.StringValue is not JSON serializable
```

```
$ ansible --version
ansible 2.7.8
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/Users/tkim/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/tkim/.local/share/virtualenvs/ansible-sskOqjbK/lib/python3.7/site-packages/ansible
  executable location = /Users/tkim/.local/share/virtualenvs/ansible-sskOqjbK/bin/ansible
  python version = 3.7.2 (default, Feb 12 2019, 08:15:36) [Clang 10.0.0 (clang-1000.11.45.5)]
```

```
$ cat /etc/ansible/ansible.cfg
[inventory]
enable_plugins = vmware_vm_inventory

```

```
$ cat dynamic.vmware.yml
plugin: vmware_vm_inventory
strict: False
hostname: vcenter01.contoso.com
username: admin_user@contoso.com
password: abcd1234
validate_certs: False
with_tags: True
```

```
$ cat requirements.txt
ansible==2.7.8
asn1crypto==0.24.0
bcrypt==3.1.6
certifi==2018.11.29
cffi==1.12.2
chardet==3.0.4
cryptography==2.6.1
idna==2.8
Jinja2==2.10
lxml==4.3.1
MarkupSafe==1.1.1
nsx-policy-python-sdk==2.3.0.0.2.11053635
nsx-python-sdk==2.3.0.0.2.11053635
paramiko==2.4.2
pyasn1==0.4.5
pycparser==2.19
PyNaCl==1.3.0
pyOpenSSL==19.0.0
pyvmomi==6.7.1.2018.12
PyYAML==3.13
requests==2.21.0
six==1.12.0
suds-jurko==0.6
urllib3==1.24.1
vapi-client-bindings==1.5.0
vapi-common-client==2.10.2
vapi-runtime==2.10.2
vmc-app-python-sdk==2.3.0.0.2.11053635
vmc-client-bindings==1.3.1
Werkzeug==0.14.1

```
Any thoughts?

**Comment by mrthomaskim** [comment]
Also tried using python 2.7 and i get similar error
```
$ ansible-inventory  --list -i dynamic.vmware.yml -vvv
ansible-inventory 2.7.8
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/Users/tkim/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Library/Python/2.7/site-packages/ansible
  executable location = /usr/local/bin/ansible-inventory
  python version = 2.7.10 (default, Aug 17 2018, 19:45:58) [GCC 4.2.1 Compatible Apple LLVM 10.0.0 (clang-1000.0.42)]
Using /etc/ansible/ansible.cfg as config file
Parsed /Users/tkim/git/ccc/sre/ansible/dynamic.vmware.yml inventory source with vmware_vm_inventory plugin
ERROR! Unexpected Exception, this is probably a bug: (vim.CustomFieldsManager.StringValue) {
   dynamicType = <unset>,
   dynamicProperty = (vmodl.DynamicProperty) [],
   key = 202,
   value = 'DEV'
} is not JSON serializable
the full traceback was:

Traceback (most recent call last):
  File "/usr/local/bin/ansible-inventory", line 118, in <module>
    exit_code = cli.run()
  File "/Library/Python/2.7/site-packages/ansible/cli/inventory.py", line 181, in run
    results = self.dump(results)
  File "/Library/Python/2.7/site-packages/ansible/cli/inventory.py", line 199, in dump
    results = json.dumps(stuff, cls=AnsibleJSONEncoder, sort_keys=True, indent=4)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/__init__.py", line 250, in dumps
    sort_keys=sort_keys, **kw).encode(obj)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/encoder.py", line 209, in encode
    chunks = list(chunks)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/encoder.py", line 434, in _iterencode
    for chunk in _iterencode_dict(o, _current_indent_level):
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/encoder.py", line 408, in _iterencode_dict
    for chunk in chunks:
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/encoder.py", line 408, in _iterencode_dict
    for chunk in chunks:
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/encoder.py", line 408, in _iterencode_dict
    for chunk in chunks:
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/encoder.py", line 408, in _iterencode_dict
    for chunk in chunks:
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/encoder.py", line 332, in _iterencode_list
    for chunk in chunks:
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/encoder.py", line 442, in _iterencode
    o = _default(o)
  File "/Library/Python/2.7/site-packages/ansible/parsing/ajson.py", line 66, in default
    value = super(AnsibleJSONEncoder, self).default(o)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/encoder.py", line 184, in default
    raise TypeError(repr(o) + " is not JSON serializable")
TypeError: (vim.CustomFieldsManager.StringValue) {
   dynamicType = <unset>,
   dynamicProperty = (vmodl.DynamicProperty) [],
   key = 202,
   value = 'DEV'
} is not JSON serializable
```

**Comment by mrthomaskim** [comment]
Hmm after few digging, i found this: https://github.com/ansible/ansible/issues/36915#issuecomment-467400417

**Comment by Akasurde** [comment]
@mrthomaskim This issue is related to Ansible and not Pyvmomi. This issue is resolved in Ansible 2.8 (devel as of 8 March 2019)(https://github.com/ansible/ansible/pull/52642) and Ansible 2.7 (https://github.com/ansible/ansible/pull/53047)

Please try above mentioned versions of Ansible and let me know if it works for you. Thanks.

**Comment by mrthomaskim** [comment]
Thanks. I can confirm it works with Ansible 2.8, devel. I have not tried it with 2.7.9 however
